Module Not Found message check does not match in Yarn v2 PnP

The "Module Not Found" message check assumes that the `First line is "Cannot find module 'foo'"`: https://github.com/jchip/optional-require/blob/28e81595f24a6adee271f6751a5b809ff2e8a08e/index.js#L5-L9

This is not true in Yarn v2 PnP, which throws a different error when a module is not found. The error message will look like any of the following:
```txt
Your application tried to access foo (a peer dependency) ...

Your application tried to access foo. While ...

Your application tried to access foo, but ...
```

We can add support for Yarn v2 PnP by also checking for any of the following in `findModuleNotFound`:
```js
msg.includes(` ${name} `)
|| msg.includes(` ${name}. `)
|| msg.includes(` ${name}, `)
```

--- a/index.js
+++ b/index.js
@@ -3,9 +3,21 @@
 const assert = require("assert");
 
 function findModuleNotFound(err, name) {
-  // First line is "Cannot find module 'foo'"
+  // Check the first line of the error message
   const msg = err.message.split("\n")[0];
-  return msg && msg.includes(`'${name}'`);
+  return msg && (
+    // Check for "Cannot find module 'foo'"
+    msg.includes(`'${name}'`)
+    // Check for "Your application tried to access foo (a peer dependency) ..." (Yarn v2 PnP)
+    // https://github.com/yarnpkg/berry/blob/e81dc0d29bb2f41818d9c5c1c74bab1406fb979b/packages/yarnpkg-pnp/sources/loader/makeApi.ts#L680
+    || msg.includes(` ${name} `)
+    // Check for "Your application tried to access foo. While ..." (Yarn v2 PnP)
+    // https://github.com/yarnpkg/berry/blob/e81dc0d29bb2f41818d9c5c1c74bab1406fb979b/packages/yarnpkg-pnp/sources/loader/makeApi.ts#L704
+    || msg.includes(` ${name}. `)
+    // Check for "Your application tried to access foo, but ..." (Yarn v2 PnP)
+    // https://github.com/yarnpkg/berry/blob/e81dc0d29bb2f41818d9c5c1c74bab1406fb979b/packages/yarnpkg-pnp/sources/loader/makeApi.ts#L718
+    || msg.includes(` ${name}, `)
+  );
 }
 
 function _optionalRequire(callerRequire, resolve, path, message) {

